webapplication/aoi/management/commands/_notebook.py

```diff
@@ -106,15 +109,22 @@ def execute_notebook(self):
         for container in exited_containers:
             attrs = Container.container_attrs(container)
             request = Request.objects.get(pk=attrs['pk'])
+            request_transaction = request.transactions.first()
+            request_transaction.completed = True
+            request_transaction.user.on_hold -= abs(request_transaction.amount)
             if attrs['exit_code'] == 0:
                 logger.info(f"Notebook in container {container.name} executed successfully")
                 request.calculated = True
                 request.save(update_fields=['calculated'])
+                request_transaction.user.balance -= abs(request_transaction.amount)
             else:
                 request.finished_at=localtime()
                 request.save(update_fields=['finished_at'])
                 logger.error(f"Execution container: {container.name}: exit code: {attrs['exit_code']},"
                              f"logs: {attrs['logs']}")
```

Comment:
I believe I forgot that we need to have a rolled_back field in Transaction that should be set to true in this case.